src/treq/test/test_content.py

```diff
@@ -173,3 +184,21 @@ def test_text_content_default_encoding_no_header(self):
         self.protocol.connectionLost(Failure(ResponseDone()))
 
         self.assertEqual(self.successResultOf(d), u'\xa1')
+
+    def test_text_content_unicode_headers(self):
+        """
+        Header parsing is robust against unicode header names and values.
+        """
+        self.response.headers = Headers({
+            b'Content-Type': [
```

Comment:
RFC 7230 has this to say:

>    Historically, HTTP has allowed field content with text in the
>    ISO-8859-1 charset [ISO-8859-1], supporting other charsets only
>    through use of [RFC2047] encoding.  In practice, most HTTP header
>    field values use only a subset of the US-ASCII charset [USASCII].
>    Newly defined header fields SHOULD limit their field values to
>    US-ASCII octets.  A recipient SHOULD treat other octets in field
>    content (obs-text) as opaque data.

RFC 2047 encoding looks like this:

> From: =?US-ASCII?Q?Keith_Moore?= <moore@cs.utk.edu>

As noted, this is not very common in actual use for HTTP.